### ISSUE TITLE
Join the `apollo-link-rest` party?

### ISSUE THREAD

**Issue by loganpowell** [dacz/apollo-bridge-link#17]
Just thinking that the premise of this library overlaps nicely with [apollo-link-rest](https://github.com/apollographql/apollo-link-rest) as you've mentioned in [your article](https://medium.com/@dadc/using-graphql-client-with-rest-api-9c332e5c8eb3). Wouldn't you serve the broader community even greater by contributing your insight and experience there?

**Comment by dacz** [comment]
Hi @loganpowell , thanks for this comment. I know about this project (and as you devised from my article I'm big Apollo fan). I started my project before the apollo-link-rest was ready because I needed the solution.

I had not so much time to dive deeper into apollo-link-rest. But on the first glimpse I see the differences there. Apollo-link-rest is focused to REST and does incredible magic by using directives in the schema.

My apollo-bridge-link takes slightly different approach. It is small abstraction around the principle of resolvers object and let you do whatever you want to. I was playing with possibility to GraphQL query the filesystem and I will probably need to solve the merging data from one REST service and one GraphQL server - merge to one schema (schema stitching not possible here). All of this is possible to be done with bridge - not so magically easy but I'd say more straightforward for me.

My plan here is to combine this library (and others) and prepare something helpful for developers who start with GraphQL or want to develop against GraphQL but are facing the "other" world.

In every case I'd be really happy to cooperate with Apollo team and find a time to look into the rest-link and maybe help there. They do greeeeeaaaat job.

**Comment by loganpowell** [comment]
@dacz Thank you for the prompt and detailed follow up! For a beginner, would you recommend `apollo-bridge-link` or 'apollo-link-rest`?

**Comment by loganpowell** [comment]
Sorry, I meant to also ask you: I really like your concept of being able to treat a local instance of GraphQL as if it were a remote server and then - if the time comes that the remote REST API becomed GraphQL compliant - being able to switch out `apollo-bridge-link` with a regular `apollo-client` and just delete local code. Is `apollo-link-bridge` better set up for that than `apollo-link-rest`?

**Comment by dacz** [comment]
@loganpowell Thank you for your follow up, too.

To your question what to choose - if you have medium or larger project with possibility to have GraphQL server in the future, I'd go with the `bridge` for sure. The main reason is that the philosophy of `bridge` is to mirror the _usual_ (maybe even _standardized_) GraphQL flow (request, resolvers, context, ...).

The way the server works is the way how `bridge` works. You will stay in the same mental model - client that knows only schema, GraphQL compliant endpoint, resolvers object, dataloaders, ...

I'd recommend to play with the bridge example. See how the code is shared with the server - so you can develop "as having the server" and then build `bridge-link` and demonstrate your proof-of-concept. You will get into working with context and dataloaders, which are pretty awesome piece of code from Facebook - and demonstrated in the example, too.

I plan to write some more articles about bridge and preparing small tool to help with creating resolvers object. I hope to get access to the AWS AppSync so I can test it and write howto about how to use it with bridge, too.

In short I'm pretty much excited by GraphQL, working with it intensively for nearly 2 years.

I'm happy to help you if you will have any questions or problems.

**Comment by loganpowell** [comment]
That's is what I had hoped! Let's hear what the `link-rest` response is and I will close this issue once there's a new trace.

Thank you @dacz!
